code/game/objects/items/RPD.dm

```diff
@@ -275,6 +309,11 @@ GLOBAL_LIST_INIT(transit_tube_recipes, list(
 	return ..()
 
 /obj/item/pipe_dispenser/pre_attack_secondary(obj/machinery/atmospherics/target, mob/user, params)
+	if(istype(target, /obj/machinery/air_sensor))
+		if(do_after(user, destroy_speed, target))
+			qdel(target)
+			return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```

Comment:
```suggestion
		if(!do_after(user, destroy_speed, target))
			return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
		
		qdel(target)
		return SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN
```

We don't want to do anything if they move away.

Reply:
An early return.  Nice